azurerm/helpers/azure/batch_pool.go

```diff
@@ -191,6 +191,22 @@ func FlattenBatchPoolCertificateReferences(armCertificates *[]batch.CertificateR
 	return output
 }
 
+// FlattenBatchPoolContainerConfiguration flattens a Batch pool container configuration
+func FlattenBatchPoolContainerConfiguration(armContainerConfiguration *batch.ContainerConfiguration) interface{} {
+
+	result := make(map[string]interface{})
+
+	if armContainerConfiguration == nil {
+		return nil
+	}
+
+	if armContainerConfiguration.Type != nil {
+		result["type"] = *armContainerConfiguration.Type
+	}
+
+	return result
```

Comment:
I think this needs to be:
```suggestion
	return []interface{}{result}
```
to fix:

```
------- Stdout: -------
=== RUN   TestAccDataSourceAzureRMBatchPool_complete
=== PAUSE TestAccDataSourceAzureRMBatchPool_complete
=== CONT  TestAccDataSourceAzureRMBatchPool_complete

------- Stderr: -------
panic: container_configuration: '': source data must be an array or slice, got map
```